{
  "id": "CVE-2021-26957",
  "assigner": "cve@mitre.org",
  "Published": "2021-02-09T22:15:14.023Z",
  "Modified": "2021-02-18T14:13:11.34Z",
  "last-modified": "2021-02-18T14:13:11.34Z",
  "summary": "An issue was discovered in the xcb crate through 2021-02-04 for Rust. It has a soundness violation because there is an out-of-bounds read in xcb::xproto::change_property(), as demonstrated by a format=32 T=u8 situation where out-of-bounds bytes are sent to an X server.",
  "access": {
    "authentication": "NONE",
    "complexity": "LOW",
    "vector": "NETWORK"
  },
  "impact": {
    "availability": "PARTIAL",
    "confidentiality": "PARTIAL",
    "integrity": "PARTIAL"
  },
  "cvss": 7.5,
  "exploitabilityScore": 10.0,
  "impactScore": 6.4,
  "cvss-vector": "AV:N/AC:L/Au:N/C:P/I:P/A:P",
  "references": [
    "https://rustsec.org/advisories/RUSTSEC-2021-0019.html"
  ],
  "vulnerable_configuration": [
    "cpe:2.3:a:xcb_project:xcb:-:*:*:*:*:rust:*:*",
    "cpe:2.3:a:xcb_project:xcb:2020-12-10:*:*:*:*:rust:*:*",
    "cpe:2.3:a:xcb_project:xcb:2021-02-04:*:*:*:*:rust:*:*"
  ],
  "vulnerable_product": [
    "cpe:2.3:a:xcb_project:xcb:-:*:*:*:*:rust:*:*",
    "cpe:2.3:a:xcb_project:xcb:2020-12-10:*:*:*:*:rust:*:*",
    "cpe:2.3:a:xcb_project:xcb:2021-02-04:*:*:*:*:rust:*:*"
  ],
  "vendors": [
    "xcb_project"
  ],
  "products": [
    "xcb"
  ],
  "vulnerable_product_stems": [
    "cpe:2.3:a:xcb_project:xcb"
  ],
  "vulnerable_configuration_stems": [
    "cpe:2.3:a:xcb_project:xcb"
  ],
  "cwe": "CWE-125",
  "vulnerable_configuration_cpe_2_2": []
}